modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -247,7 +247,7 @@
 
     /** h2 redirection stub. */
     public static final Pattern H2_REDIRECTION_RULES =
-        Pattern.compile("\\s*(create\\s*table|drop\\s*table|alter\\s*table)", CASE_INSENSITIVE);
+        Pattern.compile("\\s*(create\\s*table|create\\s*index|drop\\s*table|alter\\s*table)", CASE_INSENSITIVE);
```

Comment:
possibly : create\\s*(table|index) would be more clear here?

Reply:
yep, fixed